facefusion/processors/modules/expression_restorer.py

```diff
@@ -0,0 +1,294 @@
+		clear_processor()
+	if state_manager.get_item('video_memory_strategy') == 'strict':
+		clear_face_analyser()
+		clear_content_analyser()
+		clear_face_occluder()
+
+
+def restore_expression(source_vision_frame : VisionFrame, target_face: Face, temp_vision_frame : VisionFrame) -> VisionFrame:
+	model_template = get_options('model').get('template')
+	model_size = get_options('model').get('size')
+	expression_restorer_factor = map_float(float(state_manager.get_item('expression_restorer_factor')), 0, 200, 0, 2)
+	source_vision_frame = cv2.resize(source_vision_frame, temp_vision_frame.shape[:2][::-1])
+	source_crop_vision_frame, _ = warp_face_by_face_landmark_5(source_vision_frame, target_face.landmark_set.get('5/68'), model_template, model_size)
+	temp_crop_vision_frame, affine_matrix = warp_face_by_face_landmark_5(temp_vision_frame, target_face.landmark_set.get('5/68'), model_template, model_size)
+	box_mask = create_static_box_mask(temp_crop_vision_frame.shape[:2][::-1], state_manager.get_item('face_mask_blur'), (0, 0, 0, 0))
+	crop_masks =\
+	[
+		box_mask
+	]
+
+	if 'occlusion' in state_manager.get_item('face_mask_types'):
+		occlusion_mask = create_occlusion_mask(temp_crop_vision_frame)
+		crop_masks.append(occlusion_mask)
+	source_crop_vision_frame = prepare_crop_frame(source_crop_vision_frame)
+	temp_crop_vision_frame = prepare_crop_frame(temp_crop_vision_frame)
+	temp_crop_vision_frame = apply_restore_expression(source_crop_vision_frame, temp_crop_vision_frame, expression_restorer_factor)
+	temp_crop_vision_frame = normalize_crop_frame(temp_crop_vision_frame)
+	crop_masks.append(create_face_mask(temp_crop_vision_frame))
+	crop_mask = numpy.minimum.reduce(crop_masks).clip(0, 1)
+	temp_vision_frame = paste_back(temp_vision_frame, temp_crop_vision_frame, crop_mask, affine_matrix)
+	return temp_vision_frame
+
+
+def apply_restore_expression(source_crop_vision_frame : VisionFrame, temp_crop_vision_frame : VisionFrame, expression_restorer_factor : float) -> VisionFrame:
+	frame_processor = get_processor()
+
+	with thread_semaphore():
+		feature_3d = frame_processor.get('feature_extractor').run(None,
+		{
+			'input': temp_crop_vision_frame
+		})[0]
+
+	with thread_semaphore():
+		_, _, _, source_expression, _, _ = frame_processor.get('motion_extractor').run(None,
+		{
+			'input': source_crop_vision_frame
+		})
+
+	with thread_semaphore():
+		temp_rotation, temp_scale, temp_translation, temp_expression, temp_key_points_raw, temp_key_points = frame_processor.get('motion_extractor').run(None,
+		{
+			'input': temp_crop_vision_frame
+		})
+
+	expression = source_expression * expression_restorer_factor + temp_expression * (1 - expression_restorer_factor)
+	key_points = temp_scale * (temp_key_points_raw @ temp_rotation + expression) + temp_translation
+
+	with thread_semaphore():
+		crop_vision_frame = frame_processor.get('generator').run(None,
+		{
+			'feature_3d': feature_3d,
```

Comment:
not happy with the inputs... why _3d as suffix? also key points are kinda generic and does not describe what it is

add target, source and type to naming